defender/poap/autotasks/mint-poap/index.js

```diff
@@ -1,42 +1,92 @@
+  if (!event) { throw new Error('event undefined'); }
+  const { secrets } = event;
+  if (!secrets) { throw new Error('secrets undefined'); }
+
+  const tokenType = secrets[tokenTypeSecretName];
+  const tokenAddress = secrets[tokenAddressSecretName];
+  const nftQty = ethers.BigNumber.from(1);
+  // ERC1155 Specific data
+  const nftId = ethers.BigNumber.from(secrets[nftIdSecretName]);
+  const nftData = 0x00;
+
+  if (typeof tokenType !== 'string') { throw new Error('Token type not specified') }
+  if (!ethers.utils.getAddress(tokenAddress)) { throw new Error('Valid token address not specified') };
+  if (!tokenType) { throw new Error('Token address not specified') };
+  if (!event.request?.body) { throw new Error('body undefined') }
+
+  const { address, signature, message } = event.request.body;
+  if (!address) { throw new Error('Address not specified in request') };
+  if (!signature) { throw new Error('Signature not specified in request') };
+  if (!message) { throw new Error('Message not specified in request') };
+
+  const formattedAddress = ethers.utils.getAddress(address);
   const provider = new DefenderRelayProvider(event);
   const signer = new DefenderRelaySigner(event, provider, {
     speed: 'fast',
   });
-  const { address, signature, message } = event.request.body;
+
+  // Update Chain ID from relay
+  const chainId = await signer.getChainId();
+  domain.chainId = chainId;
 
   const messageSigner = ethers.utils.verifyTypedData(domain, types, message, signature);
 
+  // Case insensitive matching of ERC20 or ERC-20 etc.
+  const erc20 = /erc-?20/gi;
+  const erc721 = /erc-?721/gi;
+  const erc1155 = /erc-?1155/gi;
+
   if (messageSigner == address) {
     let tx;
-    if (TOKEN_TYPE == TokenTypeEnum.ERC20) {
+    if (tokenType.match(erc20)) {
       const abi = ERC20Abi;
-      const contract = new ethers.Contract(TOKEN_ADDRESS, abi, signer);
-      tx = await contract.mint(address, '1');
+      const contract = new ethers.Contract(tokenAddress, abi, signer);
+      tx = await contract.mint(formattedAddress, '1');
       console.log('Minted ERC20 token with tx:', tx.hash);
-    }
-    if (TOKEN_TYPE == TokenTypeEnum.ERC1155) {
-      const abi = ERC1155Abi;
-      const contract = new ethers.Contract(TOKEN_ADDRESS, abi, signer);
-      tx = await contract.mint(address, ERC1155ID, '1');
+    } else if (tokenType.match(erc1155)) {
+      const contract = new ethers.Contract(tokenAddress, ERC1155Abi, signer);
+      const contractV2 = new ethers.Contract(tokenAddress, ERC1155AbiV2, signer);
+      let tx;
+      try {
+        // Depending on which contract was used, the mint function might require 3 or 4 parameters
+        tx = await contract.mint(formattedAddress, nftId, nftQty, nftData);
+      } catch (error) {
+        console.log(`First attempt failed with error: ${error}`)
+        tx = await contractV2.mint(formattedAddress, nftId, nftQty);
+      }
```

Comment:
Would it be possible to condense this into just one contract call and only use the abi of `ERC1155AbiV2 ` and eliminate the abi of `ERC1155Abi `? We would then always use 4 parameters and then pass in empty data for the last parameter. 

i think this should work unless there is a difference between `0x00` and `""` being passed in as the last parameter that would effect the `_mint` behavior

Reply:
Depending on which contact is deployed, the mint() will accept 3 or 4 params. I tested it in hardhat, and it reverts when sending 4 params to a 3 param function.  `keccak('mint(address,uint256,uint256,bytes)') !== keccak('mint(address,uint256,uint256)')`. The function selectors will not line up.